vowpalwabbit/parse_args.cc

```diff
@@ -1625,6 +1625,16 @@ vw* initialize_with_builder(std::unique_ptr<options_i, options_deleter_type> opt
 
     return &all;
   }
+  catch (VW::save_load_model_exception& e)
+  {
+    const bool DETETE_ALL = true, FINALIZE_REDUCTIONS=true;
+    std::stringstream better_msg;
+    better_msg << e.what() << " , model files = ";
+    for(const auto& regressor_name : all.initial_regressors)
+      better_msg << regressor_name << ", ";
+    finish(all, DETETE_ALL, !FINALIZE_REDUCTIONS);
```

Comment:
Since we are tearing down VW, all we actually need to do here is `delete &all`, all cleanup is encapsulated in the destructor. That means you don't need to conditionally turn off the finalize regressor logic.